app/helpers/sessions_helper.rb

```diff
@@ -4,16 +4,71 @@ def log_in user
     session[:user_id] = user.id
   end
 
-  def current_user
-    @current_user ||= User.find_by id: session[:user_id]
-  end
-
   def logged_in?
     current_user.present?
   end
 
+  def forget user
+    user.forget
+    cookies.delete :user_id
+    cookies.delete :remember_token
+  end
+
   def log_out
-    reset_session
+    forget current_user
+    delete_session_and_cookies
     @current_user = nil
   end
+
+  def delete_session_and_cookies
+    session.delete(:user_id)
+    session.delete(:session_token)
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end
+
+  def current_user
+    @current_user ||= find_user_from_session || find_user_from_cookies
+  end
+
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user&.remember_token
+  end
+
+  def remember_session user
+    user.remember
+    session[:session_token] = user.remember_token
+  end
+
+  private
+
+  def find_user_from_session
+    user_id = session[:user_id]
+    return nil unless user_id
+
+    user = User.find_by(id: user_id)
+    return nil unless user
+
+    session_token = session[:session_token]
+    return nil if session_token && !user.authenticated?(session_token)
+
+    user
```

Comment:
```suggestion
    user_id = session[:user_id]
    return unless user_id

    user = User.find_by(id: user_id)
    return unless user

    session_token = session[:session_token]
    user if session_token && user.authenticated?(session_token)
```
return không có giá trị chính là return nil rồi

Reply:
Fix tương tự cho logic dưới